Angle bracket invocation guide (#401)
ember-learn/guides-source: guides/v3.7.0/reference/syntax-conversion-guide.md

## Angle Bracket Syntax

> The [angle bracket invocation syntax (ABIS)](https://github.com/emberjs/rfcs/blob/master/text/0311-angle-bracket-invocation.md) is an alternative style of invoking components in templates.

Reviewer: I'm pretty sure it's the convention to capitalise the letters of a phrase you're about to create an abbreviation from 🤔 so that would be: 

```suggestion
The [Angle Bracket Invocation Syntax (ABIS)](https://github.com/emberjs/rfcs/blob/master/text/0311-angle-bracket-invocation.md) is an alternative style of invoking components in templates.
```

Author: I am going the opposite way and not use abbreviations at all. Coming back to the material and reading it, I don't think it is improved by the use of abbvs.

Reviewer: Yes I agree 👍

> The difference between ABIS and classic invocation syntax (CIS) is syntactical and does not affect the semantics of invoking a component.

Reviewer: same as above: 

```suggestion
The difference between ABIS and Classic Invocation Syntax (CIS) is syntactical and does not affect the semantics of invoking a component.
```

Classic invocation syntax may also be referred to as curly invocation syntax.

> **Classic invocation syntax:**

Reviewer: I would revert this to keep consistent with above 👍

```handlebars
{{site-header user=this.user class=(if this.user.isAdmin "admin")}}
```

> **Angle bracket invocation syntax:**

Reviewer: ```suggestion
**Angle Bracket Invocation Syntax:**
```

</SuperSelect>
```

### Positional parameters

The classic invocation syntax supports passing arguments to the component by their position.

> In the following example, `"first"` and `"second"` are positional parameters:

Reviewer 2: ```suggestion
In the following example, `"hello"` and `"world"` are positional parameters:
```
Shouldnt this be hello and world to match the example below?
Should this be

there are two ways to handle them inside the component.
One way is to individually specify what component property the positional parameter should map to.
The other way is to map all positional parameters to the `params` property and refer to them by their index.

#### Individual names

If `my-greeting` had the following implementation:

```javascript {data-filename="app/components/my-greeting.js"}
import Component from '@ember/component';

export default Component.extend({
}).reopenClass({
  positionalParams: ['greeting', 'name']
});
```

To invoke it using the angle bracket syntax, you would do the following:

```handlebars
<MyGreeting @greeting="Hello" @name="World" />
```

#### `params` array

If `my-greeting` has the following implementation:

```javascript {data-filename="app/components/my-greeting.js"}
import Component from '@ember/component';

export default Component.extend({
}).reopenClass({
  positionalParams: 'params'
});
```

```handlebars
<MyGreeting @params={{array "Hello" "World"}}>
```

### What is happening with the Classic Invocation Syntax?

> Classic invocation syntax—which uses “curlies” `{{}}` instead of angle brackets `<>`–is here to stay.

Reviewer 3: is there a reason you have "curlies" here instead of handlebars? handlebars and angle brackets seem more consistent with each other.

Author: I kept it from the original, will update.
